ChangeLog.md

```diff
@@ -20,6 +20,10 @@ See docs/process.md for more on how version tagging works.
 
 3.1.51 (in development)
 -----------------------
+- Emscripten is now more strict about handling unsupported shared library
+  inputs.  For example if you passed `/usr/lib/libz.so` emscripten used to
+  magically redirect that to its own internal libz if it could find one.
+  (#20886)
```

Comment:
Maybe this could say what a user should do if they run into this? I guess they should provide a absolute path to an emscripten dylib, or emscripten will search its own library paths if the path isn't absolute?

Reply:
You shouldn't be passing system libraries to emscripten at all.  So I guess the advice would "don't use host libraries with emscripten, you need to use emscripten-specific libraries".